CRM/Utils/Rule.php

```diff
@@ -574,7 +574,10 @@ public static function money($value) {
       return TRUE;
     }
 
-    return preg_match('/(^-?\d+\.\d?\d?$)|(^-?\.\d\d?$)/', $value) ? TRUE : FALSE;
+    // Allow values such as -0, 1.024555, -.1
+    // We need to support multiple decimal places here, not just the number allowed by locale
+    //  otherwise tax calculations break when you want the inclusive amount to be a round number (eg. £10 inc. VAT requires 8.333333333 here).
+    return preg_match('/(^-?\d+\.?\d*$)|(^-?\.\d+$)/', $value) ? TRUE : FALSE;
```

Comment:
I feel like we could @JoeMurray @KarinG any thoughts? WMF made a change much like this a long time ago due to Cryptocurrencies & some middle eastern ones & we have been running with it. From what I can tell the DB field actually drops additional decimal places if it uses DECIMAL(10,2) - so this Rule change allows us to accept more decimal places, but it will only result in change if the DB field supports it